helix-vcs/Cargo.toml

```diff
@@ -0,0 +1,23 @@
+[package]
+name = "helix-vcs"
+version = "0.6.0"
+authors = ["Blaž Hrastnik <blaz@mxxn.io>"]
+edition = "2021"
+license = "MPL-2.0"
+categories = ["editor"]
+repository = "https://github.com/helix-editor/helix"
+homepage = "https://helix-editor.com"
+
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
+
+[dependencies]
+
+git-repository = {version = "0.23.1", default-features = false}
+tokio = { version = "1", features = ["rt", "rt-multi-thread", "time", "sync", "parking_lot", "macros"] }
+similar = "2.2"
+ropey = { version = "1.5", default-features = false, features = ["simd"] }
+arc-swap = "1"
```

Comment:
With these dependencies, number of crates to build increase by ~50-60, debug build time increased by 10s (~15%) from 59s to 69s, with git-repository I would say the timing is similar or slightly slower, I get results from 65s-76s, but now there is a duplicate crate bstr that needs rebuild since one is on 1.0 (git-repository) and all others are still using 0.2 (globset, grep-regex, grep-searcher, similar).

Reply:
> I will look into PR for those things asweel if @Byron is open to that and does but require large changes

Absolutely :)! `git-repository` did have a few more feature toggles to do be able to customize dependencies, but I found it cumbersome to maintain and removed them over time. After all, I didn't need them myself.
However, I think it's the right thing to re-add them based on actual demand.

Reply:
> @Byron By the way, broken link https://github.com/Byron/gitoxide/blob/main/cargo-features.md#git-config on gitoxide readme.

Thanks for the hint, fixed in https://github.com/Byron/gitoxide/commit/b833a4d3ffa0024feaf99103768abd0d714dce77.

Reply:
I don't want this PR to get stuck in limbo so I will do the upgrade to git-repository 0.24 in a follow-up PR as soon as the MSRV PR is merged and the other dependencies have been update